### ISSUE TITLE
Option to display outer frame of viewer without the object crosshairs

### ISSUE THREAD

**Issue by xcelente** [NatronGitHub/Natron#691]
### 

- [X] I have read the [feature request guidelines](https://hackmd.io/@natron-dev-awesome/B1SW6Hbau)
- [X] I have followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTING.md) to the best of my understanding
- [X] This feature is not on the [issue tracker](https://github.com/NatronGitHub/Natron/issues?q=is%3Abug+type%3Afeature+type%3Aidea) and is not a duplicate

### Provide a description of your feature request

```markdown
I'd like to see the outline of the viewer without displaying the object crosshairs inside.
```

### Is this a realistically implementable feature?

- [X] My feature is realistically implementable
- [ ] My feature is NOT realistically implementable

### Can you contribute in creating this feature?

- [X] I am unable, or not willing to contribute to creating this feature
- [ ] I am able and willing to contribute to the creation of this feature

### Additional details

```markdown
Not an urgent need, just a nicety.
```

**Comment by devernay** [comment]
I don't understand your request, can you maybe post a picture of what you would like to see?

**Comment by xcelente** [comment]
https://imgur.com/a/D8jgyit

Multiple objects can clutter the display / distraction. But I do want to display the outer white rectangle to help visualise the final result i.e. without the border it's harder to judge distances to edge.

**Comment by xcelente** [comment]
Having thought more about this, I think it would be even better if I could overlay a grid of lines that scales with the view. The objective is to assist in the positioning of objects within the space. Not thick lines, and maybe translucent.

**Comment by Shrinks99** [comment]
Close the transform node panel that is likely open?  These crosshairs only display when actively using a node that requires them.  They'll go away once you close that panel.  The object crosshairs aren't there to visualize the center of the frame for you, they exist so you can move objects around.  They are a control and not an alignment tool.

### Panel open, crosshair on screen

![Screenshot 2021-11-05 150737](https://user-images.githubusercontent.com/5672810/140565326-73683511-78bc-4634-a963-13a527ac5639.png)


### Panel closed, crosshair is gone!

![Screenshot 2021-11-05 150825](https://user-images.githubusercontent.com/5672810/140565340-c32b316f-46fd-4531-8c6f-c571d09d5470.png)

There's already an [open issue about alignment ](https://github.com/NatronGitHub/Natron/issues/106 )as well as an [issue open for grids](https://github.com/NatronGitHub/Natron/issues/84).

Seems like this should be closed accordingly.  Hope this clears some things up for you!

**Comment by xcelente** [comment]
Thanks for the explanation.

For me, this seems a bit clumsy as I tend to keep multiple panels open. I suppose I could close them down but hmm, I dunno.

I appreciate the crosshairs are not for alignment, but the white outer frame does help with the special placement. I wasn't aware of the grid and alignment posts, sorry about that.

Btw, is there a dash address for me to make a small donation / tip to this project?

**Comment by Shrinks99** [comment]
I personally only work with more than 1 panel open when I need to copy information between nodes but that's just what I like to do, other people work differently.  Working this way ensures that only the node I am currently working on is visible in the properties panel, same for all viewer controls.

There isn't an option in the transform node to hide the crosshairs but I don't think that would do much for users that closing the panel already achieves.

Currently Natron does not accept donations.

**Comment by xcelente** [comment]
I just had a thought. Maybe the object crosshair should only be displayed when you select the node (single click) instead of its properties being open (double click(? Multi-select with shift to display multiple object crosshairs.

**Comment by Shrinks99** [comment]
So the crosshairs won't be visible when the properties panel is open (when the user wants to change node properties), and will only be visible when it's closed but selected in the graph? (When the user is moving nodes around?)  Sounds very unintuitive to me.  The reason that crosshair exists is to provide visual feedback and control of the node's transform properties.

This system currently functions as expected and is in-line with how other nodes with viewer controls & overlays function.  I firmly do not believe it should be changed.

If you don't want to look at these positioning controls anymore when working tell them to go away by closing the node's controls in the properties panel!  If you want them to quickly go away while your properties panel still has active transform nodes, hit "O" in the viewer to disable all overlays.  Note that this also removed the frame bounding box.

**Comment by xcelente** [comment]
Yeah, you may well be right, though so far I almost never use the crosshairs. I do use the 'O' but then the frame bounding box disappears which I want to keep. I don't know what else to say really, was just trying to give some feedback. Thanks.

**Comment by devernay** [comment]
I think we should display only the overlays for nodes that are open in the parameters panel AND are in the render PATH for the viewer. I thought we had that already, but maybe that was in Natron 3...

@xcelente would that help you de-clutter the viewer?

**Comment by devernay** [comment]
proposed change in #745 (corresponding to issue #744)

**Comment by devernay** [comment]
also note that minimizeing the properties panel (rather than closing it) also hides the corresponding overlay. This is the middle button in that row:
<img width="121" alt="image" src="https://user-images.githubusercontent.com/3778071/148654402-d9e79b01-c5bc-44fd-bdcb-421c6bfb6425.png">

There was a bug where the viewer would not be immediately updated, which is fixed in #745
